pkg/controller/imagepulljob/imagepulljob_controller.go

```diff
@@ -151,6 +150,17 @@ func (r *ReconcileImagePullJob) Reconcile(request reconcile.Request) (res reconc
 		return reconcile.Result{}, err
 	}
 
+	// If resourceVersion expectations have not satisfied yet, just skip this reconcile
+	resourceVersionExpectations.Observe(job)
```

Comment:
May I ask a question?
Why some workloads such as cloneSet, ImagePullJob need check whether their resourceVersion expectations  have been satisfied and some workloads have no need to check?

Reply:
Controllers like CloneSet, ImagePullJob have reentrant risk.

For example, a imagepulljob with `parallelism=1` had added the pulling task into one nodeimage in previous reconcile, if there is no `resourceVersionExpectations`, it might add the task into another nodeimage in next immediate reconcile, because the modification of the previous nodeimage has not been watched and synced into informer.